src/Database/Postgres/Temp/Internal/Core.hs

```diff
@@ -132,7 +132,7 @@ teeHandle orig f =
           modifyIORef outputRef (<>theLine)
           hPutStrLn orig theLine
 
-    res <- withAsync readerLoop $ \_ -> f writeEnd
+    res <- withAsync readerLoop $ f writeEnd
```

Comment:
The callback is now provided a handle to the background thread that's reading `readEnd`.

Previously, after `initdb` finished, the callback was closed immediately, which canceled this thread before it could read the pipe until EOF.